### ISSUE TITLE
`AmplifyForgotPassword` UI repeats input field

### ISSUE THREAD

**Issue by thathurtabit** [aws-amplify/amplify-js#8629]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line

System:
    OS: macOS 11.4
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 121.16 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.5 - ~/.nvm/versions/node/v14.15.5/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.15.5/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.15.5/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 91.0.4472.164
    Edge: 92.0.902.55
    Firefox: 89.0.2
    Safari: 14.1.1
  npmPackages:
    @apollo/client: ^3.3.20 => 3.3.20 
    @apollo/client/cache:  undefined ()
    @apollo/client/core:  undefined ()
    @apollo/client/errors:  undefined ()
    @apollo/client/link/batch:  undefined ()
    @apollo/client/link/batch-http:  undefined ()
    @apollo/client/link/context:  undefined ()
    @apollo/client/link/core:  undefined ()
    @apollo/client/link/error:  undefined ()
    @apollo/client/link/http:  undefined ()
    @apollo/client/link/persisted-queries:  undefined ()
    @apollo/client/link/retry:  undefined ()
    @apollo/client/link/schema:  undefined ()
    @apollo/client/link/utils:  undefined ()
    @apollo/client/link/ws:  undefined ()
    @apollo/client/react:  undefined ()
    @apollo/client/react/components:  undefined ()
    @apollo/client/react/context:  undefined ()
    @apollo/client/react/data:  undefined ()
    @apollo/client/react/hoc:  undefined ()
    @apollo/client/react/hooks:  undefined ()
    @apollo/client/react/parser:  undefined ()
    @apollo/client/react/ssr:  undefined ()
    @apollo/client/testing:  undefined ()
    @apollo/client/utilities:  undefined ()
    @aws-amplify/ui-react: ^1.2.3 => 1.2.4 
    @testing-library/jest-dom: ^5.11.4 => 5.14.1 
    @testing-library/react: ^11.1.0 => 11.2.7 
    @testing-library/user-event: ^12.1.10 => 12.8.3 
    @types/jest: ^26.0.15 => 26.0.23 
    @types/node: ^12.0.0 => 12.20.15 (15.12.5)
    @types/react: ^17.0.0 => 17.0.11 
    @types/react-combine-reducers: ^1.0.0 => 1.0.0 
    @types/react-dom: ^17.0.0 => 17.0.8 
    @types/react-router-dom: ^5.1.7 => 5.1.7 
    @types/styled-components: ^5.1.10 => 5.1.10 
    aws-amplify: ^4.1.1 => 4.1.2 
    graphql: ^15.5.1 => 15.5.1 (14.0.0, 0.11.7)
    quicktype: ^15.0.260 => 15.0.260 
    react: ^17.0.2 => 17.0.2 
    react-combine-reducers: ^1.1.1 => 1.1.1 
    react-dom: ^17.0.2 => 17.0.2 
    react-icons: ^4.2.0 => 4.2.0 
    react-router-dom: ^5.2.0 => 5.2.0 
    react-scripts: 4.0.3 => 4.0.3 
    styled-components: ^5.3.0 => 5.3.0 
    styled-components/macro:  undefined ()
    styled-components/native:  undefined ()
    styled-components/primitives:  undefined ()
    swears: ^0.1.0 => 0.1.0 
    typescript: ^4.1.2 => 4.3.4 (3.2.4)
    web-vitals: ^1.0.1 => 1.1.2 
  npmGlobalPackages:
    @aws-amplify/cli: 5.1.2
    npm: 6.14.11
    yarn: 1.22.10

```

</details>

### Describe the bug

Using `AmplifyForgotPassword` react UI component repeats the input fields (it should only require one field)

I.e.

```
  <AmplifyForgotPassword
    slot="forgot-password"
    headerText="Header Test"
    formFields={[{ type: "email" }]}
  />
```

Results in: 

<img width="455" alt="amp-ui-sign-in" src="https://user-images.githubusercontent.com/1598625/126777085-d93ec821-fb07-4e55-9e37-a515c420c946.png">

### Expected behavior

It should only product one email input, not three.

### Reproduction steps

* "aws-amplify": "^4.1.1",
* "react": "^17.0.2",
* "@aws-amplify/ui-react": "^1.2.3",

```
<AmplifyAuthContainer>
          <AmplifyAuthenticator>
           
            <AmplifyForgotPassword
              slot="forgot-password"
              headerText="Header Text"
              formFields={[{ type: "email" }]}
            />
            
          </AmplifyAuthenticator>
        </AmplifyAuthContainer>

```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

const awsmobile = {
    "aws_project_region": "us-east-1",
    "aws_appsync_graphqlEndpoint": "https://bvodoevelzfa5c3q4gejukdlk4.appsync-api.us-east-1.amazonaws.com/graphql",
    "aws_appsync_region": "us-east-1",
    "aws_appsync_authenticationType": "API_KEY",
    "aws_appsync_apiKey": "******",
    "aws_cognito_identity_pool_id": "us-east-1:d0b6abe4-c565-4b16-ad8f-b269d810aa39",
    "aws_cognito_region": "us-east-1",
    "aws_user_pools_id": "us-east-1_reNumZDm5",
    "aws_user_pools_web_client_id": "1odf8h7epf0kjjn019vnv732p7",
    "oauth": {}
};

export default awsmobile;

**Comment by thathurtabit** [comment]
As far as I can tell, the error seems to be causes when React re-renders the component. Is it lacking a `key` that React uses to monitor the DOM? https://reactjs.org/docs/lists-and-keys.html#keys

**Comment by thathurtabit** [comment]
The `AmplifyForgotPassword` component also has the same error referenced here: https://github.com/aws-amplify/amplify-js/issues/5655 ("`Username` cannot be empty" - despite using `email` not `username`)

<img width="452" alt="Screen Shot 2021-07-23 at 13 47 10" src="https://user-images.githubusercontent.com/1598625/126783611-bd0621b3-b251-4939-9aa2-4fe70c0bad29.png">

```
  <AmplifyForgotPassword
    slot="forgot-password"
    headerText="Header Text"
    usernameAlias="email"
    formFields={[
      {
        type: "email",
      },
    ]}
  />
```

**Comment by chrisbonifacio** [comment]
@thathurtabit I was able to reproduce this issue with `@aws-amplify/ui-react@1.2.3` but no the latest version of 1.2.7. Can you try upgrading and let me know if you still experience the issue?

**Comment by thathurtabit** [comment]
Hi @chrisbonifacio 👋 thanks for looking into this. I'm afraid though the repeating input is still there with `"@aws-amplify/ui-react": "^1.2.7",` for myself.

**Comment by chrisbonifacio** [comment]
@thathurtabit Okay, I will try again. When it happened for me on 1.2.3, it seemed to be after signup -> signin -> logout. Then when I click Reset password, I see two Email Address fields but could not figure out how to make it duplicate again after the first time.

Have you been able to identify what actions cause it to duplicate for you or does it look like the screenshot you shared from the start?

**Comment by chrisbonifacio** [comment]
The `Username cannot be empty` error message is probably because that's the wording for the `Auth.forgotPassword` method.

https://docs.amplify.aws/lib/auth/manageusers/q/platform/js#forgot-password

```js
// Send confirmation code to user's email
Auth.forgotPassword(username)
    .then(data => console.log(data))
    .catch(err => console.log(err));

**Comment by thathurtabit** [comment]
My Authentication UI is in a Modal, so it might be something to do with the UI state getting confused when then Modal is opened and closed? But I do have a separate Sign out button, so it could also be that...

**Comment by thathurtabit** [comment]
An easy way to set the validation message would be really useful. I'm not sure if that exists already or is a feature request.

**Comment by wlee221** [comment]
I believe I found the code that's causing the problem:

https://github.com/aws-amplify/amplify-js/blob/main/packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx#L86-L92

`this.newFormFields = []` should be reset to an empty array whenever this runs... 😅  I'll make a pr for this right now.

**Comment by wlee221** [comment]
For the validation message, we'll create another PR when we get the capacity. In the meantime, you can customize the message with [Translations](https://docs.amplify.aws/ui/customization/translations/q/framework/vue):

```ts
import { I18n } from "aws-amplify";
import { Translations } from "@aws-amplify/ui-components";

I18n.putVocabulariesForLanguage("en-US", {
  [Translations.EMPTY_USERNAME]: "Your email cannot be empty"
});
```

**Comment by wlee221** [comment]
Hi @thathurtabit, we just merged in fix #8633 that will be published in the next release cycle. Please let us know if the issue persists after then!

**Comment by thathurtabit** [comment]
Thank you @wlee221! ✨✨
